src/graph/backend/dnnl/op_executable.hpp

```diff
@@ -1481,21 +1494,53 @@ struct bn_folding_t : public op_executable_t {
         memory epsilon_mem = make_dnnl_memory(desc_.epsilon_desc_,
                 scratchpad.get_engine(), (void *)buf_start);
 
-        // 1. sqrt_variance = sqrt(variance + epsilon)
-        if (variance.get_engine().get_kind() == engine::kind::cpu) {
-            float *ptr = (float *)epsilon_mem.get_data_handle();
-            *ptr = desc_.epsilon_;
-        } else {
+        if (scratchpad.get_engine().get_kind() == engine::kind::gpu) {
+#if DNNL_GPU_RUNTIME != DNNL_RUNTIME_NONE \
+        && DNNL_GPU_VENDOR == DNNL_VENDOR_NVIDIA
```

Comment:
Why do we need to handle this here? I support for NVIDIA GPU, execution will run into the `execute_sycl()` function below?